pkg/genericapiserver/api/handlers/rest.go

```diff
@@ -603,33 +610,63 @@ func patchResource(
 			// 2. build a strategic merge patch from originalJS and the currentJS
 			// 3. ensure no conflicts between the two patches
 			// 4. apply the #1 patch to the currentJS object
-			currentObjectJS, err := runtime.Encode(codec, currentObject)
-			if err != nil {
-				return nil, err
-			}
-			currentPatch, err := strategicpatch.CreateStrategicMergePatch(originalObjJS, currentObjectJS, versionedObj)
-			if err != nil {
-				return nil, err
-			}
-			originalPatch, err := strategicpatch.CreateStrategicMergePatch(originalObjJS, originalPatchedObjJS, versionedObj)
-			if err != nil {
-				return nil, err
-			}
 
-			diff1 := make(map[string]interface{})
-			if err := json.Unmarshal(originalPatch, &diff1); err != nil {
+			// TODO: This should be one-step conversion that doesn't require
+			// json marshaling and unmarshaling once #39017 is fixed.
+			var currentObjMap map[string]interface{}
+			if data, err := runtime.Encode(codec, currentObject); err != nil {
 				return nil, err
+			} else {
+				currentObjMap = make(map[string]interface{})
+				if err := json.Unmarshal(data, &currentObjMap); err != nil {
+					return nil, err
+				}
 			}
-			diff2 := make(map[string]interface{})
-			if err := json.Unmarshal(currentPatch, &diff2); err != nil {
-				return nil, err
+
+			var currentPatchMap map[string]interface{}
```

Comment:
I thought about it, but it is using few variables from this and modifying few others, so it made this even more complicated...